apprise/plugins/NotifyXMPP/__init__.py

```diff
@@ -57,35 +56,20 @@ class NotifyXMPP(NotifyBase):
     # Lower throttle rate for XMPP
     request_rate_per_sec = 0.5
 
-    # The default XMPP port
-    default_unsecure_port = 5222
-
-    # The default XMPP secure port
-    default_secure_port = 5223
-
```

Comment:
I had no idea this wasn't being used.  The intention here was for the `url()` call to default upstream servers to use port `5222` for `xmpp://` calls and `5223` for `xmpps://`.  The idea here was to work simlar to how MQTT does it (see [here](https://github.com/caronc/apprise/blob/76700bfa1ddcb2812d9ed14dfa7736125bcd346e/apprise/plugins/NotifyMQTT.py#L122-L126) and [here](https://github.com/caronc/apprise/blob/76700bfa1ddcb2812d9ed14dfa7736125bcd346e/apprise/plugins/NotifyMQTT.py#L266-L269)).  But I can understand you dropped them due them not being referenced (good catch btw).

would it make more sense that `xmpp://` and `xmpps://` just saved you from doing the `:port` when it's similar to how http works (80 is always assumed for insecure and 443 for secure unless otherwise specified). Thoughts?

Reply:
Unlike HTTP, and apparently MQTT (which I don’t know about) where you have default ports, XMPP uses special DNS records called SRV records which let the admins of a service configure a very specific order and likeliness for the various hostnames/ports to be connected to, for instance for failover or for load balancing.  This makes port 5222 the very last fallback step if no SRV record was found, before giving up on connecting altogether.  slixmpp already implements the SRV specification with the right fallback ports, so it’s not useful (and is actually harmful) to do it here.

The `:port` thing is still optional after this PR, it tells slixmpp to stop doing SRV resolving and to override the connection to the hostname/port specified and no other one. If the user’s domain is using port 5222 or 5223 it will continue to connect to them, just that if the SRV tells us to use any other port it will use it instead of failing.

As for the difference between your `xmpp://` and `xmpps://` fake URI schemes, it’s also a somewhat old difference, nowadays both are always encrypted, the first one through STARTTLS and the latter through TLS from the beginning. The only benefits of doing direct TLS are to reduce the number of roundtrips by one, and to be more friendly to HTTPS proxies, but there is no difference in security (in both cases the hostname to be connected to is sent over plain text, and that’s the only information that is). I would recommend to eventually let go of your `xmpps://` altogether, but slixmpp doesn’t implement [XEP-0368](https://xmpp.org/extensions/xep-0368.html) yet, which is the extension defining the same SRV dance as for STARTTLS but over direct TLS, so in the meantime it’s fine to keep both despite no difference in security.

Reply:
Thank you for such an amazing response! I really appreciate it! You're doing a great job educating me here :+1: